joern-cli/frontends/php2cpg/src/main/scala/io/joern/php2cpg/parser/PhpParser.scala

```diff
@@ -7,63 +7,115 @@ import io.joern.x2cpg.utils.ExternalCommand
 import org.slf4j.LoggerFactory
 
 import java.nio.file.Paths
+import java.util.regex.Pattern
+import scala.collection.mutable
 import scala.io.Source
 import scala.util.{Failure, Success, Try}
 
 class PhpParser private (phpParserPath: String, phpIniPath: String, disableFileContent: Boolean) {
 
   private val logger = LoggerFactory.getLogger(this.getClass)
 
-  private def phpParseCommand(filename: String): String = {
+  private def phpParseCommand(filenames: collection.Seq[String]): String = {
     val phpParserCommands = "--with-recovery --resolve-names --json-dump"
-    s"php --php-ini $phpIniPath $phpParserPath $phpParserCommands $filename"
+    val filenamesString   = filenames.mkString(" ")
+    s"php --php-ini $phpIniPath $phpParserPath $phpParserCommands $filenamesString"
   }
 
-  def parseFile(inputPath: String): Option[(PhpFile, Option[String])] = {
-    val inputFile      = File(inputPath)
-    val inputFilePath  = inputFile.canonicalPath
-    val inputDirectory = inputFile.parent.canonicalPath
-    val command        = phpParseCommand(inputFilePath)
-    ExternalCommand.run(command, inputDirectory) match {
-      case Success(output) =>
-        val content = Option.unless(disableFileContent)(inputFile.contentAsString)
-        processParserOutput(output, inputFilePath).map((_, content))
-      case Failure(exception) =>
-        logger.error(s"Failure running php-parser with $command", exception.getMessage)
-        None
+  def parseFiles(inputPaths: collection.Seq[String]): collection.Seq[(String, Option[PhpFile], String)] = {
+    val absoluteInputPath = inputPaths.map { inputPath =>
+      val inputFile = File(inputPath)
+      inputFile.canonicalPath
     }
-  }
 
-  private def processParserOutput(output: Seq[String], filename: String): Option[PhpFile] = {
-    val maybeJson = linesToJsonValue(output, filename)
-    maybeJson.flatMap(jsonValueToPhpFile(_, filename))
+    val command = phpParseCommand(absoluteInputPath)
+
+    val (returnValue, output) = ExternalCommand.runWithMergeStdoutAndStderr(command, ".")
```

Comment:
I don't know if this is something that could ever happen with how PhpParser handles output, but I'm wondering if it's possible to end up in a situation where an error occurs, printing something to stderr, but we still get JSON output (perhaps with error nodes or something like that) and the error and JSON output end up interleaved in some way. This could result in the JSON parsing to fail.

If we handle stdout and stderr as separate streams, then we could prepend stderr to stdout which would result in the error output being included in `infoLines` below.

With that being said, I might be misunderstanding the purpose of merging these, so please ignore if my comment doesn't make sense.

Reply:
The problem is that the php parser gives information about the parsed file on stderr. This includes the file name and log messages about parsing errors and the actual json is provided on stdout.
This was easy to handle so far because only one file was given to the php parser. So the relation between file name, log messages and the json ouput was clear.
Now that we give multiple files as input this is not the case anymore, although the php parser seems to preserve the order, i did not want to (fully) rely on it. To a certain amount our code still does because we assume a certain timing between stderr and stdout which ensures that the file name comes first on stderr and than comes the json on stdout.
The hole interface provided by the php parsers is pretty bad to consume jsons for multiple file but sadly that is what it is. 
To improve we will like need to provide a PR upstream.

Reply:
Ah, I see. In that case I don't see any alternative to this implementation until this is improved upstream